src/sage/combinat/k_regular_sequence.py

```diff
@@ -918,6 +979,367 @@ def _n_to_index_(self, n):
+                the corresponding list of column indices (i.e., arguments to
+                the current left vector valued sequence).
+            """
+            d = len(seq(0)) + len(lines)
+
+            for m_indices in cantor_product(xsrange(n_max), repeat=d, min_slope=1):
+                # Iterate over all increasing lists of length d consisting
+                # of non-negative integers less than `n_max`.
+
+                U = Matrix(domain, d, d, [values(m, lines) for m in m_indices]).transpose()
+                try:
+                    return U.inverse(), m_indices
+                except ZeroDivisionError:
+                    pass
+            else:
+                raise RuntimeError('no inverse submatrix found')
+
+        def linear_combination_candidate(t_L, r_L, lines):
+            r"""
+                Based on an invertible submatrix of ``A`` as described in the
+                algorithm section of the docstring, find a candidate for a
+                linear combination of the rows of ``A`` yielding the subsequence
+                with parameters ``t_L`` and ``r_L``, i.e.,
+                `m \mapsto f(k**t_L * m + r_L)`.
+            """
+            iU, m_indices = some_inverse_U_matrix(lines)
+            X_L = vector(f(k**t_L * m + r_L) for m in m_indices)
+            return X_L * iU
+
+        def verify_linear_combination(t_L, r_L, linear_combination, lines):
+            r"""
+                Determine whether the subsequence with parameters ``t_L`` and
+                ``r_L``, i.e., `m \mapsto f(k**t_L * m + r_L)`, is the linear
+                combination ``linear_combination`` of the current vector valued
+                sequence.
+            """
+            return all(f(k**t_L * m + r_L) ==
+                       linear_combination * vector(values(m, lines))
+                       for m in xsrange(0, (n_max - r_L) // k**t_L + 1))
+
+        class NoLinearCombination(ValueError):
+            pass
+
+        def find_linear_combination(t_L, r_L, lines):
+            linear_combination = linear_combination_candidate(t_L, r_L, lines)
+            if not verify_linear_combination(t_L, r_L, linear_combination, lines):
+                raise NoLinearCombination
+            return linear_combination
+
+        if seq(0).is_zero():
+            left = None
+        else:
+            try:
+                left = vector(find_linear_combination(0, 0, []))
+            except NoLinearCombination:
+                left = None
+
+        to_branch = []
+        lines = []
+
+        def include(line):
+            to_branch.append(line)
+            lines.append(line)
+            t, r = line
+            logger.info('including f_{%s*m+%s}', k**t, r)
+
+        if left is None:
+            line_L = (0, 0)  # entries (t, r) --> k**t * m + r
+            include(line_L)
+            left = vector((len(seq(0)) + len(lines)-1)*(zero,) + (one,))
```

Comment:
At this point, it should be clear that `lines` has exactly one element (that has just been added to the empty list), so we could simplify the code here.
```suggestion
            assert len(lines) == 1
            left = vector(len(seq(0))*(zero,) + (one,))
```